src/hotspot/share/classfile/verificationType.cpp

```diff
@@ -64,7 +64,7 @@ bool VerificationType::resolve_and_check_assignability(InstanceKlass* klass, Sym
     }
   }
 
-  if (this_class->access_flags().is_value_class()) return false;
```

Comment:
This line looks wrong to me. If the method is called with name and from_name being identical, and the corresponding class being a primitive class, the method returns false.
This particular scenario is handled in the caller when this method is invoked from VerificationType::is_reference_assignable_from(), but it not handled when called from SystemDictionaryShared::check_verification_constraints().
This problem is not due to the proposed change, the previous code testing for a value class suffers from the same issue.

Reply:
I agree.  See changes to this code in commit 2.   Thanks for looking at this PR.